### ISSUE TITLE
Function to import tools from URL

### ISSUE THREAD

**Issue by keiranmraine** [rabix/composer#393]
Firstly sorry if I've failed to find the relevant bit of documentation.

It appears you can modify the tool cwl path after using a local copy of a repository but it would be really good if there was a specific 'Tool from URL' option.

For example public tools registered on dockstore.org have the CWL served via https:

```
https://dockstore.org:8443/api/ga4gh/v2/tools/quay.io%2Fwtsicgp%2Fdockstore-cgpmap%2Fcgpmap-cramOut/versions/3.0.3/plain-CWL/descriptor/%2Fcwls%2Fcgpmap-cramOut.cwl
```

[cgpmap-cramOut](https://dockstore.org/containers/quay.io/wtsicgp/dockstore-cgpmap/cgpmap-cramOut)

**Comment by kaushik-work** [comment]
Hello @keiranmraine , do I understand correctly that your use case is to include a tool from a tool repository somewhere? In such a case you want the UI to automatically add the $import or $include directive, is that it? Thanks!

**Comment by denis-yuen** [comment]
The idea would be to import a tool from a tool repository/URL. 
If the tool has imports or includes, it would need to resolve them appropriately relative to the URL. 

For example, the above URL references 

```
doc:
  $include: includes/doc.yml
```

so the compose would also need to pull in 
```
https://dockstore.org:8443/api/ga4gh/v2/tools/quay.io%2Fwtsicgp%2Fdockstore-cgpmap%2Fcgpmap-cramOut/versions/3.0.3/plain-CWL/descriptor/includes/doc.yml
```

(tagging @agduncan94 )

**Comment by kaushik-work** [comment]
@denis-yuen thank you! 

I'm trying to think what the behavior would be when someone types $import or $include in the `code` panel. If we resolved these the moment the user switched to the `Visual Editor` panel then the raw code would change. This might be acceptable behavior given how RC is designed.

**Comment by ivanbatic** [comment]
When you use `$import`, `$include` and `$mixin` keys, files which they reference are fetched and embedded into the document that is loaded into the editor. You are likely to see this behavior in action when opening a local workflow that references other files for it's steps.

This also works for remote urls, and [resolver](https://github.com/rabix/composer/blob/master/electron/src/schema-salad-resolver/schema-salad-resolver.ts) will make http calls if necessary, meaning that the functionality to load a file from URL exists, it just doesn't have a UI.

**Comment by keiranmraine** [comment]
@kaushik-work, yes, as @denis-yuen has summarised (better than I probably would have).
